### PR TITLE
feat(mailer): add support for categories

### PR DESCRIPTION
**Pull request by regarin**
This PR adds support for adding categories to mails

### CODE REVIEW

File: spec/dummy/spec/lib/send_grid_mailer/definition_spec.rb [hunk 1]
```diff
@@ -120,6 +120,14 @@
     end
   end
 
+  describe "#add_category" do
+    it "adds categories to mail object" do
+      subject.add_category("category1")
+      subject.add_category("category2")
```
Comment (monkeyci) [inline spec/dummy/spec/lib/send_grid_mailer/definition_spec.rb:126]: Name your test subject if you need to reference it explicitly.

File: spec/dummy/spec/lib/send_grid_mailer/definition_spec.rb [hunk 2]
```diff
@@ -120,6 +120,14 @@
     end
   end
 
+  describe "#add_category" do
+    it "adds categories to mail object" do
+      subject.add_category("category1")
```
Comment (monkeyci) [inline spec/dummy/spec/lib/send_grid_mailer/definition_spec.rb:125]: Name your test subject if you need to reference it explicitly.

File: lib/send_grid_mailer/definition.rb [hunk 3]
```diff
@@ -72,6 +73,11 @@ def add_header(key, value)
       personalization.add_header(SendGrid::Header.new(key: key, value: value))
     end
 
+    def add_category(value)
+      return unless value
```
Comment (monkeyci) [inline lib/send_grid_mailer/definition.rb:77]: Add empty line after guard clause.

**Comment by joaquinnunez** [comment]
❤️

**Comment by ldlsegovia** [comment]
awesome :)

**Pull request merged**